modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/ObjectValidationRuleEngineExceptionMapper.java

```diff
@@ -26,14 +29,39 @@
 public class ObjectValidationRuleEngineExceptionMapper
 	extends BaseExceptionMapper<ObjectValidationRuleEngineException> {
 
+	public ObjectValidationRuleEngineExceptionMapper(Language language) {
+		_language = language;
+	}
+
 	@Override
 	protected Problem getProblem(
 		ObjectValidationRuleEngineException
 			objectValidationRuleEngineException) {
 
-		return new Problem(
-			Response.Status.BAD_REQUEST,
-			objectValidationRuleEngineException.getMessage());
+		if (objectValidationRuleEngineException.getMessageKey() == null)
+
+			return new Problem(
+				Response.Status.BAD_REQUEST,
+				_language.get(
+					_acceptLanguage.getPreferredLocale(),
+					objectValidationRuleEngineException.getMessage()));
+			else
+
+			return new Problem(
+				Response.Status.BAD_REQUEST,
+				_language.get(
+					_acceptLanguage.getPreferredLocale(),
+					objectValidationRuleEngineException.getMessageKey()));
```

Comment:
@diogomelo09, I believe this approach below is more simple, what do you think?

````
String messageKey = objectValidationRuleEngineException.getMessageKey();

if (messageKey == null) {
	messageKey = objectValidationRuleEngineException.getMessage();
}

return new Problem(
	Response.Status.BAD_REQUEST,
	_language.get(
		_acceptLanguage.getPreferredLocale(), messageKey));
````

Reply:
Just did it!! :smile: